ph5/utilities/utmtolatlong.py

```diff
@@ -0,0 +1,167 @@
+                easting = float(fieldset[0])
+                northing = float(fieldset[1])
+                if easting < 0.0 or northing < 0.0:
+                    print ("Error, easting and northing must be positive,\
+                        but you used ", easting, " and ", northing)
+                    return
+                try:
+                    zone = int(fieldset[2])
+                except (RuntimeError, TypeError, NameError, ValueError):
+                    print ("Error, zone must be an integer, from 1 - 60, \
+                        but you used ", fieldset[2])
+                    return
+                if zone < 1 or zone > 60:
+                    print ("Error, zone must be from 1 - 60, but you used ",
+                           zone)
+                    return
+                hemisphere = str(fieldset[3]).upper()
+                if hemisphere != "S" and hemisphere != "N":
+                    print ("Error, zone must be either N or S for \
+northern/southern hemisphere, but you used ", hemisphere)
+                    return
+
+                utmi = utm_conversions(easting, northing, zone, hemisphere)
+                lon, lat = utmi.utm_to_latlong()
+                print ("easting=%.1f, northing=%.1f, zone=%d, hemisphere=%s"
+                       % (easting, northing, zone, hemisphere))
+                print ("Lon = %.7f, Lat = %.7f" % (lon, lat))
+                outstr = "%.7f, %.7f" % (lon, lat) + "\n"
+                output.write(outstr)
+
+        finally:
+            dater.close()
+            output.close()
+
+    def doinline(self, args):
+
+        easting = float(args.easting)
+        northing = float(args.northing)
+        if easting < 0.0 or northing < 0.0:
+            print ("Error, easting and northing must be >0, but you used ",
+                   easting, " and ", northing)
+            return
+        try:
+            zone = int(args.zone)
+        except (RuntimeError, TypeError, NameError, ValueError):
+            print ("Error, zone must be an integer from 1-60, but you used ",
+                   args.zone)
+            return
+        if zone < 1 or zone > 60:
+            print ("Error, zone must be from 1 - 60, but you used ", zone)
+            return
+        hemisphere = str(args.side).upper()
+        if hemisphere != "S" and hemisphere != "N":
+            print ("Error, zone must be either N or S for northern/southern \
+hemisphere, but you used ", hemisphere)
+            return
+
+        print ("easting=%.1f, northing=%.1f, zone=%d, hemisphere=%s"
+               % (easting, northing, zone, hemisphere))
+        utmi = utm_conversions(easting, northing, zone, hemisphere)
+        lon, lat = utmi.utm_to_latlong()
+        result = "Lon = %.7f, Lat = %.7f" % (lon, lat)
+        print (result)
+        print ("***************************************")
+
+
+def main():
+
+    desc = 'Convert UTM locations to longitudes, latitudes, \

```

Comment:
I'm noticing these `\^M` carriage-return characters. Are you working on a Windows machine?

Please set the file encoding to UTF-8 and line-endings for new files to Unix in your IDE. This is so that text files are saved in a format that is not specific to the Windows OS and most easily shared across heterogeneous developer desktops.